src/implementations/Backend.Fx.EfCorePersistence/EfFlush.cs

```diff
@@ -106,7 +106,14 @@ private void UpdateTrackingProperties(string userId, DateTime utcNow)
                      .ForAll(entry =>
                      {
                          EntityEntry aggregateRootEntry = FindAggregateRootEntry(DbContext.ChangeTracker, entry);
-                         if (aggregateRootEntry.State == EntityState.Unchanged) aggregateRootEntry.State = EntityState.Modified;
+                         if (aggregateRootEntry == null)
+                         {
+                             throw new InvalidOperationException($"Could not find aggregate root of {entry.Entity.GetType().Name}[{entry.Entity?.Id}]");
```

Comment:
what about moving this throw to the FindAggregateRootEntry method, instead of `return null` in the end? And attribute it with `[NotNull]`? Would be even more intuitive. And since my personal convention is _"Find"=maybe null, "Get"=must not be null_ I'd even rename it to `GetAggregateRootEntry()`

Reply:
Thanks. I moved the null check to the method and renamed it accordingly.